sensor/common/detector/baseline/baseline_bench_test.go

```diff
@@ -0,0 +1,279 @@
+package baseline
+
+import (
+	"flag"
+	"fmt"
+	"runtime"
+	"testing"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/protocompat"
+)
+
+var (
+	benchMax = flag.Bool("bench.max", false, "Run maximum scale benchmarks (300k containers)")
+)
+
+// benchmarkMemoryUsage is a parameterized benchmark function
+func benchmarkMemoryUsage(b *testing.B, evaluatorFactory func() Evaluator, baselines []*storage.ProcessBaseline, scenarioName string, showDeduplication bool) {
+	runtime.GC()
+	runtime.GC()
+
+	var m1, m2 runtime.MemStats
+	runtime.ReadMemStats(&m1)
+
+	evaluator := evaluatorFactory()
+	for _, baseline := range baselines {
+		evaluator.AddBaseline(baseline)
+	}
+
+	runtime.GC()
+	runtime.GC()
+	runtime.ReadMemStats(&m2)
```

Comment:
Optional: I didn't have good results with this approach of measuring memory consumption. I got much more stable/repeatable results with disabling the GC temporary. 


```go
import (
    "runtime"
    "runtime/debug"
    // ... other imports
)

// benchmarkMemoryUsage is a parameterized benchmark function
func benchmarkMemoryUsage(b *testing.B, evaluatorFactory func() Evaluator, baselines []*storage.ProcessBaseline, scenarioName string, showDeduplication bool) {
    // Disable automatic GC during benchmark
    oldGCPercent := debug.SetGCPercent(-1)
    defer debug.SetGCPercent(oldGCPercent) // Restore original GC setting
    ...
```